src/Spice86.Core/Emulator/VM/Breakpoint/EmulatorBreakpointsManager.cs

```diff
@@ -115,4 +124,125 @@ public void CheckExecutionBreakPoints() {
             _cycleBreakPoints.TriggerMatchingBreakPoints(cycles);
         }
     }
+
+
+    public SerializableUserBreakpointCollection CreateSerializableBreakpoints() {
+        var serializableBreakpoints = new SerializableUserBreakpointCollection();
+        AddBreakpointsToCollection(serializableBreakpoints, _executionBreakPoints.SerializableBreakpoints);
+        AddBreakpointsToCollection(serializableBreakpoints, InterruptBreakPoints.SerializableBreakpoints);
+        AddBreakpointsToCollection(serializableBreakpoints, _cycleBreakPoints.SerializableBreakpoints);
+        AddBreakpointsToCollection(serializableBreakpoints, MemoryReadWriteBreakpoints);
+        AddBreakpointsToCollection(serializableBreakpoints, IoReadWriteBreakpoints.SerializableBreakpoints);
+        return serializableBreakpoints;
+    }
+
+    private void AddBreakpointsToCollection(SerializableUserBreakpointCollection serializableBreakpoints,
+        AddressReadWriteBreakpoints memoryReadWriteBreakpoints) {
+        if (!memoryReadWriteBreakpoints.SerializableBreakpoints.Any()) {
+            return;
+        }
+
+        foreach (SerializableUserBreakpoint breakpoint in GroupConsecutiveBreakpoints(
+            memoryReadWriteBreakpoints.SerializableBreakpoints).Distinct()) {
+            serializableBreakpoints.Breakpoints.Add(breakpoint);
+        }
+    }
+
+    private static IEnumerable<SerializableUserBreakpoint> GroupConsecutiveBreakpoints(
+        IEnumerable<AddressBreakPoint> memoryBreakpoints) {
+
+        if (!memoryBreakpoints.Any()) {
+            yield break;
+        }
+
+        foreach (var group in memoryBreakpoints
+            .Select((bp, index) => new {
+                Breakpoint = bp,
+                Index = index,
+                GroupKey = bp.Address - index
```

Comment:
Are you sure this works if input list is not sorted? I would think grouping logic depends on address ordering is the same as index ordering
Consecutive addresses should have consecutive indexes